### PR TITLE
JS/1926-Nearest Exit From Entrance in Maze

### PR DESCRIPTION
**Pull request by codyconway77**

### CODE REVIEW

**Review by jmlawrence — APPROVED** [review]
Awesome work!

File: JS/1926. Nearest Exit From Entrance In Maze.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            // 1. Set this step location to a "wall" to represent we have already visited
+            maze[ny][nx] = "+";
+            // 2. Add this valid step to the back of the queue
+            // This allows for steps in the queue with a lower step count to be processed first,
+            // Thus ensuring we always return the shortest path.
+            q.unshift([ny, nx, nextStep]);
+        }
+    }
+    // All valid steps have been removed from the queue and no exits have been found
+    return -1;
```
Comment (toni783) [inline JS/1926. Nearest Exit From Entrance In Maze.js:76]: nit. One optimization that can be made here is to return -1 for the default use cases where the loops won't be needed i.e  array with only two elements won't have an exit

**Review by toni783 — APPROVED** [review]
LGTM 👍🏼

File: JS/1926. Nearest Exit From Entrance In Maze.js (hunk 1 above)
Comment (codyconway77) [reply JS/1926. Nearest Exit From Entrance In Maze.js:76]: Good catch. Helps with time complexity for some base cases

**Pull request merged**